pytorch_lightning/trainer/callback_hook.py

```diff
@@ -87,20 +95,44 @@ def on_validation_epoch_start(self):
         for callback in self.callbacks:
             callback.on_validation_epoch_start(self, self.lightning_module)
 
-    def on_validation_epoch_end(self):
-        """Called when the epoch ends."""
+    def on_validation_epoch_end(self, outputs):
+        """Called when the epoch ends.
+
+        Args:
+            outputs: List of outputs on each `val` epoch
+        """
         for callback in self.callbacks:
-            callback.on_validation_epoch_end(self, self.lightning_module)
+            if self._is_param_in_hook_signature(callback.on_validation_epoch_end, "outputs"):
+                callback.on_validation_epoch_end(self, self.get_model(), outputs)
+            else:
+                warning_cache.warn(
+                    "`Callback.on_validation_epoch_end` signature has changed in v1.3."
+                    " `outputs` parameter has been added."
+                    " Support for the old signature will be removed in v1.5", DeprecationWarning
+                )
+                callback.on_validation_epoch_end(self, self.get_model())
 
     def on_test_epoch_start(self):
         """Called when the epoch begins."""
         for callback in self.callbacks:
             callback.on_test_epoch_start(self, self.lightning_module)
 
-    def on_test_epoch_end(self):
-        """Called when the epoch ends."""
+    def on_test_epoch_end(self, outputs):
+        """Called when the epoch ends.
+
+        Args:
+            outputs: List of outputs on each `test` epoch
+        """
         for callback in self.callbacks:
-            callback.on_test_epoch_end(self, self.lightning_module)
+            if self._is_param_in_hook_signature(callback.on_test_epoch_end, "outputs"):
+                callback.on_test_epoch_end(self, self.get_model(), outputs)
+            else:
+                warning_cache.warn(
+                    "`Callback.on_test_epoch_end` signature has changed in v1.3."
+                    " `outputs` parameter has been added."
+                    " Support for the old signature will be removed in v1.5", DeprecationWarning
```

Comment:
Needs a deprecation test

Reply:
Added. I see that you have added a utility for `no_deprecated_call`, could use it. Will add that once yours get merged or else follow up PR.

Reply:
Feel free to copy/paste it and use it here, if you don't want to wait

Reply:
Umm, there's an issue with that, right? It will throw an Assertion Error in case of any Deprecation Warning even if not related to the tests.

Reply:
True. You can add a `match` argument to check for the correct deprecation warning

Reply:
Yup, updated!